test/e2e/utils/recordedit-helpers.js

```diff
@@ -321,7 +347,51 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
                     });
                 });
             }
+            
+            if (jsonCols.length > 0) {
+                describe("JSON fields, ", function () {
+                    it("should show textarea input for JSON datatype and then set the value", function() {
+                        jsonCols.forEach(function(c) {
+                        chaisePage.recordEditPage.getTextAreaForAcolumn(c.name, recordIndex).then(function(jsonTxtArea) {
+                            if (jsonTxtArea) {
+                                expect(true).toBeDefined();
+                                jsonTxtArea.column = c;
+                                
+                                JSONDataTypeFields.push(jsonTxtArea);
+                                var value = getRecordValue(c.name);
 
+                                if (value != undefined) {
+                                    expect(jsonTxtArea.getAttribute('value')).toBe(value, colError(c.name , "Doesn't have the expected value."));
+                                }              
+                            }
+                            else{
+                                expect(undefined).toBeDefined();
+                            }
+                        });
+                    });
+                 });
+                 
+                 it("should only allow valid JSON values", function(){
+                     jsonCols.forEach(function(c) {
+                         chaisePage.recordEditPage.getTextAreaForAcolumn(c.name, recordIndex).then(function(jsonTxtArea) {
+                             for (i = 0; i < JSONTestParams.length; i++) {
+                                 jsonTxtArea.clear();
+                                 (function(input){
+                                     c._value = input;
+                                     jsonTxtArea.sendKeys(input);
+                                     chaisePage.recordEditPage.getJSONInputErrorMessage(jsonTxtArea, 'json').then(function(error){
+                                         if(error){
```

Comment:
I understand that you are doing this so that _if_ it does error out, the test will fail. It just seems weird that you are testing the lack of an error appearing.

This is fine as is, unless you can think of a way to verify that the input values are not throwing an error (using an expectation).